src/main.cpp

```diff
@@ -66,57 +94,63 @@ main(int argc, const char** argv)
     // Parse args
     auto [verbosity] = process_arguments(argc, argv);
 
-    // Start logging
-    auto log_level = static_cast<uint8_t>(quill::LogLevel::Debug);
+    // Start logging and print build info
+    raccoon::logging::init(verbosity);
+    log_build_info();
 
-    if (verbosity <= log_level)
-        log_level -= verbosity;
-    else // protect from underflow
-        log_level = 0;
+    // Connect to redis
+    namespace utils = raccoon::utils; // TEMP: redis will be refactored into own dir
 
-    raccoon::logging::init(static_cast<quill::LogLevel>(log_level));
+    auto redis_url = utils::getenv("REDIS_URL", "127.0.0.1");
+    auto redis_port = std::stoi(utils::getenv("REDIS_PORT", "6379"));
 
-    // Print information about the build
-    log_build_info();
-    log_i(libcurl, "{}", curl_version());
+    redisContext* ctx = redisConnect(redis_url.c_str(), redis_port);
 
-    redisContext* c = redisConnect("127.0.0.1", 6379);
-    if (c == nullptr || c->err) {
-        if (c) {
-            log_e(main, "Error: %s\n", c->errstr);
+    if (ctx == nullptr || ctx->err) [[unlikely]] {
+        if (ctx) {
+            log_e(main, "Could not connect to Redis: {}\n", ctx->errstr);
         }
         else {
             log_e(main, "Can't allocate redis context\n");
         }
         return 1;
     }
-    log_d(main, "Successfully connected to redis");
 
-    raccoon::storage::DataProcessor prox(c);
+    log_i(main, "Successfully connected to redis");
+
+    raccoon::storage::DataProcessor prox(ctx);
+
+    // Init curl
+    if (curl_global_init(CURL_GLOBAL_ALL)) {
+        log_c(main, "Could not initialize cURL");
+        return 1;
+    }
+
+    // Manager testing
+    raccoon::web::RequestManager session;
 
     // Create websocket
-    auto data_cb = [&prox](
-                       raccoon::web::WebSocketConnection* conn,
-                       std::vector<uint8_t> data
-                   ) {
-        std::string string_data(data.begin(), data.end());
-        string_data.append("\0");
-
-        log_d(main, "Data: {}", string_data);
-        if (string_data == "deadbeef") {
-            std::string str =
-                R"({"type":"subscribe","channels":[{"name":"matches","product_ids":["ETH-USD"]},{"name":"level2","product_ids":["ETH-USD"]}]})";
-            std::vector<uint8_t> bytes(str.begin(), str.end());
+    auto data_cb = [&prox](auto* conn, std::vector<uint8_t> data) {
+        if (memcmp(data.data(), PROXY_FIRST_MESSAGE, PROXY_FIRST_MESSAGE_LEN) == 0)
+            [[unlikely]] // only the first message is like this
+        {
+            std::vector<uint8_t> bytes(
+                COINBASE_SUBSCRIBE_STR.begin(), COINBASE_SUBSCRIBE_STR.end()
+            );
             conn->send(std::move(bytes));
         }
-        else {
-            prox.process_incoming_data(string_data);
+        else [[likely]] {
```

Comment:
this keyword is sick wtf

Reply:
I know, right!? I was looking into using __builtin_likely and then found out about this.